libraries/src/MVC/Controller/FormController.php

```diff
@@ -541,6 +613,117 @@ protected function postSaveHook(BaseDatabaseModel $model, $validData = array())
 	{
 	}
 
+	/**
+	 * Method to save a field for inline editing.
+	 * JSON Response: {'saved': true}  if successfully saved,
+	 *                {'saved': false} otherwise.
+	 *
+	 * @param   string  $key     The name of the primary key of the URL variable.
+	 * @param   string  $urlVar  The name of the URL variable if different from the primary key (sometimes required to avoid router collisions).
+	 *
+	 * @return  void
+	 *
+	 * @since __DEPLOY_VERSION__
+	 */
+	public function saveInline(string $key = null, string $urlVar = null): void
+	{
+		// Check for request forgeries.
+		if ($this->checkToken('post', false) == false)
+		{
+			$this->app->enqueueMessage(Text::_('JINVALID_TOKEN_NOTICE'), 'warning');
+			echo new JsonResponse(null, null, true);
+			$this->app->close();
+		}
+
+		$data = $this->input->post->get('jform', array(), 'array');
+
+		$fieldName  = $this->input->post->getString('field_name');
+		$fieldGroup = $this->input->post->getString('field_group');
+
+		if ($fieldName == null || $fieldName == '')
+		{
+			echo new JsonResponse(null, 'Empty field', true);
+			$this->app->close();
+		}
+
+		// Don't accept more than one field change at a time.
+		if (count($data) > 1 || (is_array(reset($data)) && count(reset($data)) > 1))
+		{
+			echo new JsonResponse(null, 'Can change only one field at a time.', true);
+			$this->app->close();
+		}
+
+		$model = $this->getModel();
+		$table = $model->getTable();
+
+		// Determine the name of the primary key for the data.
+		if (empty($key))
+		{
+			$key = $table->getKeyName();
+		}
+
+		// To avoid data collisions the urlVar may be different from the primary key.
+		if (empty($urlVar))
+		{
+			$urlVar = $key;
+		}
+
+		$recordId = $this->input->getInt($urlVar);
+
+		// Load the form with data
+		$model->setState('article.id', $recordId);
+		$form = $model->getForm();
+		$oldData = $form->getData()->toArray();
+
+		$processData = array_merge($oldData, $data);
+
+		foreach ($processData as $key_ => $value_)
```

Comment:
Why are there underscores after the variable names?

Reply:
I have already used variable name `$key` above and wanted to reuse it. Due to its scope, it was changing the value of earlier `$key`. So, I added a `_` after these variable names